### PR TITLE
Debug flaky test

### LINKED ISSUE #4733: odo delete with wait flag tests are flaky
/kind flaky-test
/priority High

### PR DESCRIPTION
**Pull request by feloy**
**What type of PR is this?**
> Uncomment only one ` /kind` line, and delete the rest.
> For example, `> /kind bug` would simply become: `/kind bug`

/kind flake

**What does this PR do / why we need it**:

**Which issue(s) this PR fixes**:

Fixes #4733 

**PR acceptance criteria**:

- [ ] Unit test 

- [ ] Integration test 

- [ ] Documentation 

- [ ] Update changelog

- [ ] I have read the [test guidelines](https://github.com/openshift/odo/blob/master/docs/dev/test-architecture.adoc)

**How to test changes / Special notes to the reviewer**:

### CODE REVIEW

**Comment by feloy** [comment]
```
36mINFO[0m[2021-05-20T19:19:28Z] [ssh:Fedora-32-minikube]     [90m/home/fedora/amqp_ci_rcv_odo-minikube-pr-build_PR_4736/repo/tests/integration/devfile/cmd_devfile_delete_test.go:153[0m 
[36mINFO[0m[2021-05-20T19:19:28Z] [ssh:Fedora-32-minikube]                     
[36mINFO[0m[2021-05-20T19:19:28Z] [ssh:Fedora-32-minikube]     [91mresource service/mjolte should be deleted 
[36mINFO[0m[2021-05-20T19:19:28Z] [ssh:Fedora-32-minikube]     Expected        
[36mINFO[0m[2021-05-20T19:19:28Z] [ssh:Fedora-32-minikube]         <string>: NAME     TYPE        CLUSTER-IP      EXTERNAL-IP   PORT(S)    AGE 
[36mINFO[0m[2021-05-20T19:19:28Z] [ssh:Fedora-32-minikube]         mjolte   ClusterIP   -IP REDACTED-   <none>        3000/TCP   43s 
[36mINFO[0m[2021-05-20T19:19:28Z] [ssh:Fedora-32-minikube]                     
```

File: tests/helper/helper_kubectl.go [hunk 1]
```diff
@@ -212,8 +212,10 @@ func (kubectl KubectlRunner) WaitAndCheckForTerminatingState(resourceType, names
 }
 
 func (kubectl KubectlRunner) VerifyResourceDeleted(ri ResourceInfo) {
```
Comment (valaparthvi) [inline tests/helper/helper_kubectl.go:214]: We will need to add one for OcRunner as well, no?
https://github.com/openshift/odo/blob/main/tests/helper/helper_oc.go#L586

**Review by valaparthvi — COMMENTED** [review]
One question. LGTM otherwise.

File: tests/helper/helper_kubectl.go (hunk 1 above)
Comment (feloy) [reply tests/helper/helper_kubectl.go:214]: Yes, that would be more consistent (I didn't see any error on OpenShift for this test)

File: tests/helper/helper_kubectl.go [hunk 2]
```diff
@@ -212,8 +212,10 @@ func (kubectl KubectlRunner) WaitAndCheckForTerminatingState(resourceType, names
...
-	session := CmdRunner(kubectl.path, "get", ri.ResourceType, "--namespace", ri.Namespace)
-	Eventually(session).Should(gexec.Exit(0))
-	output := string(session.Wait().Out.Contents())
-	Expect(output).NotTo(ContainSubstring(ri.ResourceName))
+	getOutput := func() string {
+		session := CmdRunner(kubectl.path, "get", ri.ResourceType, "--namespace", ri.Namespace)
+		Eventually(session).Should(gexec.Exit(0))
+		return string(session.Wait().Out.Contents())
+	}
+	Eventually(getOutput, 3*time.Second, 500*time.Millisecond).ShouldNot(ContainSubstring(ri.ResourceName), fmt.Sprintf("resource %s/%s should be deleted", ri.ResourceType, ri.ResourceName))
```
Comment (mik-dass) [inline tests/helper/helper_kubectl.go:220]: This function is called to check `odo delete -w` and thus we shouldn't wait and use eventually here as the resource should be deleted by then.

Comment (feloy) [reply tests/helper/helper_kubectl.go:220]: I think that, actually, the resources are not always deleted when the `odo delete -w` command returns: https://github.com/openshift/odo/issues/4733#issuecomment-847792991

And it seems not possible to make `odo` use the Kubernetes API to wait the deletion of the dependent resources, as the OpenShift RBAC policy forbids `odo` to add the `blockOwnerDeletion=true` field to the dependent resources, when ran as `developer`.

Comment (mik-dass) [reply tests/helper/helper_kubectl.go]: > And it seems not possible to make odo use the Kubernetes API to wait the deletion of the dependent resources, as the OpenShift RBAC policy forbids odo to add the blockOwnerDeletion=true field to the dependent resources, when ran as developer.

OK, please leave a comment in the code for now and raise a new issue for someone to look into it later.

File: tests/helper/helper_kubectl.go (hunk 1 above)
Comment (feloy) [reply tests/helper/helper_kubectl.go:218]: I've also changed on OcRunner. Thanks

File: tests/helper/helper_kubectl.go (hunk 2 above)
Comment (feloy) [reply tests/helper/helper_kubectl.go]: I've reopened #4593  and added comments on it

**Review by mik-dass — APPROVED** [review]
/lgtm

**Comment by feloy** [comment]
:( The tests are still failing regularly. I'll continue to investigate to see what else can make the test fail

**Comment by feloy** [comment]
/test psi-kubernetes-integration-e2e

Passed 2x

**Comment by feloy** [comment]
/test psi-kubernetes-integration-e2e

Passed 3x

**Comment by mohammedzee1000** [comment]
Awesome

**Review by mohammedzee1000 — APPROVED** [review]
Looks good to me 
/lgtm

**Pull request merged**